File: decompiled/4.0.2/be/ehealth/businessconnector/therlink/domain/HcParty.java
package be.ehealth.businessconnector.therlink.domain;

import be.ehealth.businessconnector.therlink.exception.TherLinkBusinessConnectorException;
import be.ehealth.businessconnector.therlink.exception.TherLinkBusinessConnectorExceptionValues;
import be.fgov.ehealth.standards.kmehr.cd.v1.CDHCPARTY;
import be.fgov.ehealth.standards.kmehr.id.v1.IDHCPARTY;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import org.apache.commons.lang.builder.EqualsBuilder;

/** @deprecated */
@Deprecated
public class HcParty implements Serializable {
   private static final long serialVersionUID = 5656836929856945713L;
   private String type;
   private String applicationID;
   private String nihii;
   private String inss;
   private String hubId;
   private String cbe;
   private String name;
   private String firstName;
   private String familyName;
   private String eHP;
   private List<IDHCPARTY> ids = new ArrayList();
   private List<CDHCPARTY> cds = new ArrayList();

   public HcParty() {
   }

   public List<IDHCPARTY> getIds() {
      return this.ids;
   }

   public void setIds(List<IDHCPARTY> ids) {
      this.ids.clear();
      this.ids.addAll(ids);
   }

   public List<CDHCPARTY> getCds() {
      return this.cds;
   }

   public void setCds(List<CDHCPARTY> cds) {
      this.cds.clear();
      this.cds.addAll(cds);
   }

   public String getType() {
      return this.type;
   }

   public void setType(String type) {
      this.type = type;
   }

   /** @deprecated */
   @Deprecated
   public String getNihii() {
      return this.nihii;
   }

   /** @deprecated */
   @Deprecated
   public void setNihii(String nihii) {
      this.nihii = nihii;
   }

   /** @deprecated */
   @Deprecated
   public String getInss() {
      return this.inss;
   }

   /** @deprecated */
   @Deprecated
   public void setInss(String inss) {
      this.inss = inss;
   }

   /** @deprecated */
   @Deprecated
   public String getHubId() {
      return this.hubId;
   }

   /** @deprecated */
   @Deprecated
   public void setHubId(String hubId) {
      this.hubId = hubId;
   }

   /** @deprecated */
   @Deprecated
   public void setEHP(String ehp) {
      this.eHP = ehp;
   }

   /** @deprecated */
   @Deprecated
   public String getEHP() {
      return this.eHP;
   }

   /** @deprecated */
   @Deprecated
   public String getCbe() {
      return this.cbe;
   }

   /** @deprecated */
   @Deprecated
   public void setCbe(String cbe) {
      this.cbe = cbe;
   }

   public String getName() {
      return this.name;
   }

   public void setName(String name) {
      this.name = name;
   }

   public String getFirstName() {
      return this.firstName;
   }

   public void setFirstName(String firstName) {
      this.firstName = firstName;
   }

   public String getFamilyName() {
      return this.familyName;
   }

   public void setFamilyName(String familyName) {
      this.familyName = familyName;
   }

   public boolean equals(Object obj) {
      if (this == obj) {
         return true;
      } else if (obj == null) {
         return false;
      } else if (this.getClass() != obj.getClass()) {
         return false;
      } else {
         HcParty other = (HcParty)obj;
         return (new EqualsBuilder()).append(this.cds, other.getCds()).append(this.ids, other.getIds()).append(this.familyName, other.getFamilyName()).append(this.firstName, other.getFirstName()).append(this.name, other.getName()).append(this.type, other.getType()).isEquals();
      }
   }

   public int hashCode() {
      int prime = true;
      int result = 1;
      result = 31 * result + (this.applicationID == null ? 0 : this.applicationID.hashCode());
      result = 31 * result + (this.cbe == null ? 0 : this.cbe.hashCode());
      result = 31 * result + (this.cds == null ? 0 : this.cds.hashCode());
      result = 31 * result + (this.eHP == null ? 0 : this.eHP.hashCode());
      result = 31 * result + (this.familyName == null ? 0 : this.familyName.hashCode());
      result = 31 * result + (this.firstName == null ? 0 : this.firstName.hashCode());
      result = 31 * result + (this.hubId == null ? 0 : this.hubId.hashCode());
      result = 31 * result + (this.ids == null ? 0 : this.ids.hashCode());
      result = 31 * result + (this.inss == null ? 0 : this.inss.hashCode());
      result = 31 * result + (this.name == null ? 0 : this.name.hashCode());
      result = 31 * result + (this.nihii == null ? 0 : this.nihii.hashCode());
      result = 31 * result + (this.type == null ? 0 : this.type.hashCode());
      return result;
   }

   public String toString() {
      StringBuilder builder = new StringBuilder();
      builder.append("HcParty ").append("[");
      builder.append("hubId=").append(this.hubId).append(", ");
      builder.append("type=").append(this.type).append(", ");
      builder.append("nihii=").append(this.nihii).append(", ");
      builder.append("inss=").append(this.inss).append("]");
      return builder.toString();
   }

   /** @deprecated */
   @Deprecated
   public String getApplicationID() {
      return this.applicationID;
   }

   /** @deprecated */
   @Deprecated
   public void setApplicationID(String value) {
      this.applicationID = value;
   }

   public static class Builder {
      private HcParty hcp = new HcParty();

      public Builder() {
      }

      public Builder withFirstName(String value) {
         this.hcp.setFirstName(value);
         return this;
      }

      public Builder withFamilyName(String value) {
         this.hcp.setFamilyName(value);
         return this;
      }

      public Builder withName(String value) {
         this.hcp.setName(value);
         return this;
      }

      public Builder withInss(String value) {
         this.hcp.setInss(value);
         return this;
      }

      public Builder withCbe(String value) {
         this.hcp.setCbe(value);
         return this;
      }

      public Builder withNihii(String value) {
         this.hcp.setNihii(value);
         return this;
      }

      public Builder withType(String value) {
         this.hcp.setType(value);
         return this;
      }

      public Builder withApplicationID(String string) {
         this.hcp.setApplicationID(string);
         return this;
      }

      public HcParty build() throws TherLinkBusinessConnectorException {
         this.validateNameConstraint();
         this.validateAtLeastOneNihiiOrInss();
         this.validateForApplicationRule();
         return this.hcp;
      }

      private void validateForApplicationRule() throws TherLinkBusinessConnectorException {
         if (this.hcp.getApplicationID() != null && (!this.hcp.getType().equalsIgnoreCase("application") || this.hcp.getNihii() != null || this.hcp.getInss() != null)) {
            throw new TherLinkBusinessConnectorException(TherLinkBusinessConnectorExceptionValues.HCP_NOT_VALID, new Object[]{"If HcParty has an applicationID, then type should be 'application' and niss neither nihii should be filled in"});
         }
      }

      private void validateAtLeastOneNihiiOrInss() throws TherLinkBusinessConnectorException {
         if (this.hcp.getNihii() == null && this.hcp.getInss() == null && this.hcp.getApplicationID() == null && this.hcp.getEHP() == null) {
            String msg = "At least nihii or inss should be defined";
            throw new TherLinkBusinessConnectorException(TherLinkBusinessConnectorExceptionValues.HCP_NOT_VALID, new Object[]{msg});
         }
      }

      private void validateNameConstraint() throws TherLinkBusinessConnectorException {
         if (!this.hasName() && !this.hasCompleteName() && this.hasAtLeastOneName()) {
            throw new TherLinkBusinessConnectorException(TherLinkBusinessConnectorExceptionValues.VALIDATION_ERROR, new Object[]{"Hcp should have a firstName and a FamilyName, (X)OR a name and nothing else"});
         }
      }

      private boolean hasAtLeastOneName() {
         return this.hcp.getFirstName() != null || this.hcp.getFamilyName() != null && this.hcp.getName() != null;
      }

      private boolean hasCompleteName() {
         return this.hcp.getFirstName() != null && this.hcp.getFamilyName() != null && this.hcp.getName() == null;
      }

      private boolean hasName() {
         return this.hcp.getFirstName() == null && this.hcp.getFamilyName() == null && this.hcp.getName() != null;
      }

      public Builder withEhp(String string) {
         this.hcp.setEHP(string);
         return this;
      }
   }
}

File: src/main/kotlin/org/taktik/connector/business/therlink/domain/HcParty.kt
package org.taktik.connector.business.therlink.domain

import be.fgov.ehealth.standards.kmehr.cd.v1.CDHCPARTY
import be.fgov.ehealth.standards.kmehr.id.v1.IDHCPARTY
import org.apache.commons.lang.builder.EqualsBuilder
import org.taktik.connector.business.therlink.exception.TherLinkBusinessConnectorException
import org.taktik.connector.business.therlink.exception.TherLinkBusinessConnectorExceptionValues

import java.io.Serializable
import java.util.ArrayList

class HcParty : Serializable {
	var type: String? = null


	@get:Deprecated("")
	@set:Deprecated("")
	var applicationID: String? = null


	@get:Deprecated("")
	@set:Deprecated("")
	var nihii: String? = null


	@get:Deprecated("")
	@set:Deprecated("")
	var inss: String? = null


	@get:Deprecated("")
	@set:Deprecated("")
	var hubId: String? = null


	@get:Deprecated("")
	@set:Deprecated("")
	var cbe: String? = null
	var name: String? = null
	var firstName: String? = null
	var familyName: String? = null


	@get:Deprecated("")
	@set:Deprecated("")
	var ehp: String? = null
	val ids = ArrayList<IDHCPARTY>()
	val cds = ArrayList<CDHCPARTY>()

	fun getIds(): List<IDHCPARTY> {
		return this.ids
	}

	fun setIds(ids: List<IDHCPARTY>) {
		this.ids.clear()
		this.ids.addAll(ids)
	}

	fun getCds(): List<CDHCPARTY> {
		return this.cds
	}

	fun setCds(cds: List<CDHCPARTY>) {
		this.cds.clear()
		this.cds.addAll(cds)
	}

	override fun equals(obj: Any?): Boolean {
		if (this === obj) {
			return true
		} else if (obj == null) {
			return false
		} else if (this.javaClass != obj.javaClass) {
			return false
		} else {
			val other = obj as HcParty?
			return EqualsBuilder().append(this.cds, other!!.getCds()).append(this.ids, other.getIds()).append(this.familyName, other.familyName).append(this.firstName, other.firstName).append(this.name, other.name).append(this.type, other.type).isEquals
		}
	}

	override fun hashCode(): Int {
		var result = 1
		result = 31 * result + if (this.applicationID == null) 0 else this.applicationID!!.hashCode()
		result = 31 * result + if (this.cbe == null) 0 else this.cbe!!.hashCode()
		result = 31 * result + if (this.cds == null) 0 else this.cds.hashCode()
		result = 31 * result + if (this.ehp == null) 0 else this.ehp!!.hashCode()
		result = 31 * result + if (this.familyName == null) 0 else this.familyName!!.hashCode()
		result = 31 * result + if (this.firstName == null) 0 else this.firstName!!.hashCode()
		result = 31 * result + if (this.hubId == null) 0 else this.hubId!!.hashCode()
		result = 31 * result + if (this.ids == null) 0 else this.ids.hashCode()
		result = 31 * result + if (this.inss == null) 0 else this.inss!!.hashCode()
		result = 31 * result + if (this.name == null) 0 else this.name!!.hashCode()
		result = 31 * result + if (this.nihii == null) 0 else this.nihii!!.hashCode()
		result = 31 * result + if (this.type == null) 0 else this.type!!.hashCode()
		return result
	}

	override fun toString(): String {
		val builder = StringBuilder()
		builder.append("HcParty ").append("[")
		builder.append("hubId=").append(this.hubId).append(", ")
		builder.append("type=").append(this.type).append(", ")
		builder.append("nihii=").append(this.nihii).append(", ")
		builder.append("inss=").append(this.inss).append("]")
		return builder.toString()
	}

	class Builder {
		private val hcp = HcParty()

		fun withFirstName(value: String?): HcParty.Builder {
			this.hcp.firstName = value
			return this
		}

		fun withFamilyName(value: String?): HcParty.Builder {
			this.hcp.familyName = value
			return this
		}

		fun withName(value: String?): HcParty.Builder {
			this.hcp.name = value
			return this
		}

		fun withInss(value: String?): HcParty.Builder {
			this.hcp.inss = value
			return this
		}

		fun withCbe(value: String?): HcParty.Builder {
			this.hcp.cbe = value
			return this
		}

		fun withNihii(value: String?): HcParty.Builder {
			this.hcp.nihii = value
			return this
		}

		fun withType(value: String?): HcParty.Builder {
			this.hcp.type = value
			return this
		}

		fun withApplicationID(string: String?): HcParty.Builder {
			this.hcp.applicationID = string
			return this
		}

		@Throws(TherLinkBusinessConnectorException::class)
		fun build(): HcParty {
			this.validateNameConstraint()
			this.validateAtLeastOneNihiiOrInss()
			this.validateForApplicationRule()
			return this.hcp
		}

		@Throws(TherLinkBusinessConnectorException::class)
		private fun validateForApplicationRule() {
			if (this.hcp.applicationID != null && (!this.hcp.type!!.equals("application", ignoreCase = true) || this.hcp.nihii != null || this.hcp.inss != null)) {
				throw TherLinkBusinessConnectorException(TherLinkBusinessConnectorExceptionValues.HCP_NOT_VALID, "If HcParty has an applicationID, then type should be 'application' and niss neither nihii should be filled in")
			}
		}

		@Throws(TherLinkBusinessConnectorException::class)
		private fun validateAtLeastOneNihiiOrInss() {
			if (this.hcp.nihii == null && this.hcp.inss == null && this.hcp.applicationID == null && this.hcp.ehp == null) {
				val msg = "At least nihii or inss should be defined"
				throw TherLinkBusinessConnectorException(TherLinkBusinessConnectorExceptionValues.HCP_NOT_VALID, msg)
			}
		}

		@Throws(TherLinkBusinessConnectorException::class)
		private fun validateNameConstraint() {
			if (!this.hasName() && !this.hasCompleteName() && this.hasAtLeastOneName()) {
				throw TherLinkBusinessConnectorException(TherLinkBusinessConnectorExceptionValues.VALIDATION_ERROR, "Hcp should have a firstName and a FamilyName, (X)OR a name and nothing else")
			}
		}

		private fun hasAtLeastOneName(): Boolean {
			return this.hcp.firstName != null || this.hcp.familyName != null && this.hcp.name != null
		}

		private fun hasCompleteName(): Boolean {
			return this.hcp.firstName != null && this.hcp.familyName != null && this.hcp.name == null
		}

		private fun hasName(): Boolean {
			return this.hcp.firstName == null && this.hcp.familyName == null && this.hcp.name != null
		}

		fun withEhp(string: String): HcParty.Builder {
			this.hcp.ehp = string
			return this
		}
	}

	companion object {
		private const val serialVersionUID = 1L
	}
}
